newroutes/homepage.js

```diff
@@ -0,0 +1,73 @@
+const autoBind = require("auto-bind")
+const express = require("express")
+
+// Import middleware
+const { BadRequestError } = require("@errors/BadRequestError")
+
+const {
+  attachReadRouteHandlerWrapper,
+  attachWriteRouteHandlerWrapper,
+} = require("@middleware/routeHandler")
+
+const { UpdateHomepageSchema } = require("@validators/RequestSchema")
+
+class HomepageRouter {
+  constructor({ homepagePageService }) {
+    this.homepagePageService = homepagePageService
+    // We need to bind all methods because we don't invoke them from the class directly
+    autoBind(this)
+  }
+
+  // Read homepage index file
+  async readHomepage(req, res) {
+    const { accessToken } = req
+
+    const { siteName } = req.params
+
+    const readResp = await this.homepagePageService.read({
+      siteName,
+      accessToken,
+    })
+
+    return res.status(200).json(readResp)
+  }
+
+  // Update homepage index file
+  async updateHomepage(req, res) {
+    const { accessToken } = req
+
+    const { siteName } = req.params
+    const { error } = UpdateHomepageSchema.validate(req.body, {
+      allowUnknown: true,
+    })
```

Comment:
i think this is a good pattern that we should be using going forward! we validate unknown data and ensure that it's of a valid shape before passing it onto our services!